Move nevergrad to the new parser (#908)
facebookresearch/hydra: website/docs/plugins/nevergrad_sweeper.md

To run hyperparameter search and look for the best parameters for this function, clone the code and run the following command in the `plugins/hydra_nevergrad_sweeper` directory:
```bash
python example/my_app.py.py -m
```

You can also override the search space parametrization:
```bash

> python example/my_app.py.py -m \
> db=mnist,cifar batch_size=4,8,16 \
> 'lr=tag(log, interval(0.001, 1))' 'dropout=interval(0,1)'

Reviewer: formatting (and full --multirun instead of -m to make it clear).
```suggestion
python example/my_app.py.py --multirun db=mnist,cifar batch_size=4,8,16 \
'lr=tag(log, interval(0.001, 1))' 'dropout=interval(0,1)'
```
Also, can you add a word about quoting the parameters?

You can also override the search space parametrization:
```bash

> python example/dummy_training.py -m \

Reviewer: can  you also rename dummy_training to my_app.py to be consistent with the rest of the examples?

Author: done :)

[HYDRA] Sweep output dir: multirun/2020-03-04/17-53-29
[HYDRA] Launching 10 jobs locally
[HYDRA] 	#0 : db=mnist batch_size=8 lr=0.032 dropout=0.5
[__main__][INFO] - my_app.py(dropout=0.500, lr=0.032, db=mnist, batch_size=8) = 5.258
[HYDRA] 	#1 : db=mnist batch_size=16 lr=0.035 dropout=0.714
[__main__][INFO] - my_app.py(dropout=0.714, lr=0.035, db=mnist, batch_size=16) = 13.469
[HYDRA] 	#2 : db=cifar batch_size=8 lr=0.053 dropout=0.408
[__main__][INFO] - my_app.py(dropout=0.408, lr=0.053, db=cifar, batch_size=8) = 4.145
[HYDRA] 	#3 : db=cifar batch_size=8 lr=0.012 dropout=0.305
[__main__][INFO] - my_app.py(dropout=0.305, lr=0.012, db=cifar, batch_size=8) = 4.133
[HYDRA] 	#4 : db=mnist batch_size=4 lr=0.030 dropout=0.204
[__main__][INFO] - my_app.py(dropout=0.204, lr=0.030, db=mnist, batch_size=4) = 1.216
```

and the final 2 evaluations look like this:
```text
[HYDRA] 	#8 : db=mnist batch_size=4 lr=0.094 dropout=0.381
[__main__][INFO] - my_app.py(dropout=0.381, lr=0.094, db=mnist, batch_size=4) = 1.077
[HYDRA] 	#9 : db=mnist batch_size=4 lr=0.094 dropout=0.381

> [__main__][INFO] - my_app.py(dropout=0.381, lr=0.094, db=mnist, batch_size=4) = 1.077

Reviewer: the output really contains `my_app.py` ?

Author: 😂 that is a copy paste oversight. will fix

## Defining the parameters

> The plugin supports 2 types of parameters: [Choices](https://facebookresearch.github.io/nevergrad/parametrization_ref.html#nevergrad.p.Choice) and [Scalars](https://facebookresearch.github.io/nevergrad/parametrization_ref.html#nevergrad.p.Scalar). They can be defined either through config file or commandline override.

Reviewer: It's not the plugin, it's Nevergrad, right?

Author: Here's the list of params nevergrad supports :https://facebookresearch.github.io/nevergrad/parametrization_ref.html#parameters

Our plugins covers half of them. For future improvements we could look into support more :)

Reviewer: roger.

## Defining the parameters

The plugin supports 2 types of parameters: [Choices](https://facebookresearch.github.io/nevergrad/parametrization_ref.html#nevergrad.p.Choice) and [Scalars](https://facebookresearch.github.io/nevergrad/parametrization_ref.html#nevergrad.p.Scalar). They can be defined either through config file or commandline override.

### Defining through commandline override

> Hydra 1.0 provides a override parser that support rich syntax. More documentation can be found in ([OverrideGrammer/Basic](../advanced/override_grammar/basic.md)) and ([OverrideGrammer/Extended](../advanced/override_grammar/extended.md)). We recommend you go through them first before proceeding with this doc.

Reviewer: ```suggestion
Hydra provides a override parser that support rich syntax. More documentation can be found in ([OverrideGrammer/Basic](../advanced/override_grammar/basic.md)) and ([OverrideGrammer/Extended](../advanced/override_grammar/extended.md)). We recommend you go through them first before proceeding with this doc.
```

    is multiplicative. 
  - `integer`: set to `true` for integers (favor floats over integers whenever possible)

Providing only `lower` and `upper` bound will set the initial value to the middle of the range, and the step to a sixth of the range.
**Note**: unbounded scalars (scalars with no upper and/or lower bounds) can only be defined through a config file.

> ### Defining through commandline override

Reviewer: I think the section about command line should come before the section about config files.
Most users will be defining sweeps through the command line.

Author: sounds good, moved :)

Hydra 1.0 provides a override parser that support rich syntax. More documentation can be found in ([OverrideGrammer/Basic](../advanced/override_grammar/basic.md)) and ([OverrideGrammer/Extended](../advanced/override_grammar/extended.md)). We recommend you go through them first before proceeding with this doc.

#### Choices
To override a field with choices:
```commandline
key=1,5
key=shuffle(range(1, 8)) # shuffled range treated as Choice
key=range(1,5) # Cho (upper bound - lower bound <= 6)
```

You can tag an override with ```ordered``` to indicate it's a [```TransitionChoice```](https://facebookresearch.github.io/nevergrad/parametrization_ref.html#nevergrad.p.TransitionChoice)
```commandline
key=tag(ordered, choice(1,2,3))
```

#### Scalar
```commandline

> key=range(1,8) # Scalar bounds cast to be integer

Author: The cast happens when we transfer the override to ng params, I see why this is confusing, will fix the wording :)

> key=float(range(1,8))  # Scalar bounds to be float

Reviewer: ```suggestion
key=float(range(1,8))  # Scalar bounds cast to a float
```

> key=interval(1,12)

Reviewer: ```suggestion
key=interval(1,12)   # Interval are float by default
```
